### DISCUSSION TITLE
Check for dependency only for a specific version

### DISCUSSION THREAD

**Discussion by iMonZ** [Homebrew/discussions#980] [Everyday usage]
I would like to fix a dependency for a formula, but the dependency was renamed!

So the formula works on every macOS version but only after 11 the dependency was renamed!

Does someone know how I can fix this issue that after 11 it checks for another dep?

**Comment by gromgit** [comment]
Untested:
```
on_macos do
  if MacOS.version <= :catalina
    depends_on "old_dep"
  else
    depends_on "new_dep"
  end
end
```

**Reply by SMillerDev** [reply]
Seeing how it can only be a system dependency based on the description, this might be more in line with the idea:

```ruby
dependency_path_var = "/some/path"
on_macos do
  if MacOS.version <= :catalina
    dependency_path_var = "/some/other/path"
  end
end
```

**Reply by iMonZ** [reply]
> Seeing how it can only be a system dependency based on the description, this might be more in line with the idea:
> 
> ```ruby
> dependency_path_var = "/some/path"
> on_macos do
>   if MacOS.version <= :catalina
>     dependency_path_var = "/some/other/path"
>   end
> end
> ```

Thanks I will try that

**Reply by iMonZ** [reply]
> Seeing how it can only be a system dependency based on the description, this might be more in line with the idea:
> 
> ```ruby
> dependency_path_var = "/some/path"
> on_macos do
>   if MacOS.version <= :catalina
>     dependency_path_var = "/some/other/path"
>   end
> end
> ```

Hey Sorry for bad explaining on my side I mean a formula dependency

**Reply by SMillerDev** [reply]
In that case the initial answer is correct. But if the homebrew formula containing the dependency was renamed that means it'll be renamed on all macOS versions.
